sematic/resolvers/local_resolver.py

```diff
@@ -420,10 +463,26 @@ def _add_edge(self, edge: Edge):
 
     def _make_run(self, future: AbstractFuture) -> Run:
         """Create a run for give future."""
-        run = make_run_from_future(future)
+        run = make_run_from_future(future=future)
         run.root_id = self._root_future.id
+        run.cache_key = self._get_cache_key(future=future)
         return run
 
+    def _get_cache_key(self, future: AbstractFuture) -> Optional[str]:
+        """
+        Computes a cache key for the specified Future, if configured.
+        """
+        if self._cache_namespace is None or not future.props.cache:
+            logger.debug("Future %s is not configured to use the cache", future.id)
+            return None
+
+        cache_key = get_future_cache_key(
```

Comment:
If your caching function uses `git`, this will not work remotely, as the git context will not be available in the image. You probably need to add the (resolved) cache namespace as a new field on the `Resolution` object, so that `worker.py` can instantiate the `Resolver` with the namespace that was produced locally before the detach.

Reply:
The [git cache namespace documentation](https://github.com/sematic-ai/sematic/blob/main/sematic/caching/git.py#L11-L13) states that it cannot be used for production envs.

It's not sufficient to assume that the origin of the pipeline must be in a local env, as pipelines can be scheduled by external, cron-like production systems, such as Buildkite. For these, there is never a local dev env.

Reply:
Regardless, we should persist the cache namespace. If only for debugging purposes, and also for the UI.

Reply:
Added.